modules/apps/batch-planner/batch-planner-web/src/main/resources/META-INF/resources/js/import/ImportForm.js

```diff
@@ -60,27 +62,82 @@ function ImportForm({
 	});
 	const [formEvaluated, setFormEvaluated] = useState(false);
 	const [fileFields, setFileFields] = useState();
+	const [fileContent, setFileContent] = useState();
 	const [demoFileValues, setDemoFileValues] = useState({});
 	const [fieldsSelections, setFieldsSelections] = useState({});
 	const [mappingsToBeEvaluated, setMappingsToBeEvaluated] = useState(
 		mappedFields
 	);
+	const [fileContentPreview, setFileContentPreview] = useState([]);
 	const useTemplateMappingRef = useRef();
+	const [formIsValid, setFormIsValid] = useState(false);
+	const [csvHeaders, setCsvHeaders] = useState(true);
+
+	useEffect(() => {
+		const requiredFieldNotFilled = dbFields.required
+			? dbFields.required.some(
+					(dbField) => !fieldsSelections[dbField.name]
+			  )
+			: false;
 
-	const formIsValid = useMemo(() => {
 		if (
-			!Object.keys(fieldsSelections).length ||
-			!(dbFields.optional.length + dbFields.required.length)
+			fieldsSelections &&
+			Object.keys(fieldsSelections).length > 0 &&
+			dbFields.optional?.length > 0 &&
+			!requiredFieldNotFilled
 		) {
-			return false;
+			setFormIsValid(true);
 		}
+		else {
+			setFormIsValid(false);
+		}
+	}, [fieldsSelections, dbFields]);
 
-		const requiredFieldNotFilled = dbFields.required.some(
-			(dbField) => !fieldsSelections[dbField.name]
-		);
+	useEffect(() => {
+		setFieldsSelections({});
+	}, [csvHeaders]);
 
-		return !requiredFieldNotFilled;
-	}, [fieldsSelections, dbFields]);
+	useEffect(() => {
+		const fieldsIndex = [];
```

Comment:
The only purpose of this `useEffect` is to update the value within `fileContentPreview`.
This leads to 2 considerations: 

1. This is not the right place to do it: `ImportPreviewModal` would fit more
2. `useMemo` would have been a better choice